File: src/main/java/shukaro/questlog/net/CommonProxy.java
package shukaro.questlog.net;

import cpw.mods.fml.common.event.*;
import net.minecraft.server.MinecraftServer;
import net.minecraftforge.common.MinecraftForge;
import shukaro.questlog.QuestConfig;
import shukaro.questlog.data.BookData;
import shukaro.questlog.data.PlayerData;
import shukaro.questlog.data.QuestData;
import shukaro.questlog.Questlog;
import shukaro.questlog.event.QuestEventHandler;

import java.io.File;

public class CommonProxy
{
    public void serverStarted(FMLServerStartedEvent evt)
    {
        Questlog.playerData = new PlayerData(new File(MinecraftServer.getServer().worldServers[0].getSaveHandler().getWorldDirectory(), "questlogPlayerData.json"));
    }

    public void preInit(FMLPreInitializationEvent evt)
    {
        Questlog.logger = evt.getModLog();
        QuestConfig.initCommon(evt);
        Questlog.questData = new QuestData(new File(evt.getModConfigurationDirectory(), Questlog.modID + File.separator + "questData.json"));
        Questlog.bookData = new BookData(new File(evt.getModConfigurationDirectory(), Questlog.modID + File.separator + "bookData.json"));
        MinecraftForge.EVENT_BUS.register(new QuestEventHandler());
    }

    public void init(FMLInitializationEvent evt)
    {

    }

    public void postInit(FMLPostInitializationEvent evt)
    {

    }
}

File: src/main/kotlin/shukaro/questlog/sided/CommonProxy.kt
package shukaro.questlog.sided

import net.minecraftforge.common.config.Configuration
import net.minecraftforge.fml.common.event.*
import shukaro.questlog.Config
import shukaro.questlog.Questlog

open class CommonProxy {

    fun serverStarting(event: FMLServerStartingEvent){}
    fun postInit(event: FMLPostInitializationEvent){}
    fun imcRequest(event: FMLInterModComms){}

    fun preInit(event: FMLPreInitializationEvent) {
        Questlog.logger = event.modLog
        Questlog.config = Configuration(event.suggestedConfigurationFile)
        Config.readConfig()
    }

    fun init(event: FMLInitializationEvent) {

    }
}